exchange/src/exchange/traders/trader_manager.hpp

```diff
@@ -13,64 +12,54 @@
 namespace nutc {
 namespace manager {
 
-using trader_t = std::variant<remote_trader_t, local_trader_t, bot_trader_t>;
-
 class ClientManager {
-    std::unordered_map<std::string, trader_t> traders_{};
+    std::unordered_map<std::string, std::unique_ptr<generic_trader_t>> traders_;
 
 public:
-    [[nodiscard]] std::string
-    add_client(const trader_t& trader)
+    void
+    add_remote_trader(const std::string& user_id, const std::string& algo_id)
     {
-        std::string user_id =
-            std::visit([](auto&& trader) { return trader.get_id(); }, trader);
-        traders_.emplace(user_id, trader);
-        return user_id;
+        traders_.emplace(user_id, std::make_unique<remote_trader_t>(user_id, algo_id));
```

Comment:
you should be using `make_shared`

there are very few cases when `unique_ptr` is the right answer over `shared_ptr`